client/src/components/ui/input.tsx

```diff
@@ -4,18 +4,22 @@ import { cn } from '@/lib/classnames';
 
 export type InputProps = React.InputHTMLAttributes<HTMLInputElement>;
 
-const Input = React.forwardRef<HTMLInputElement, InputProps>(
-  ({ className, type, ...props }, ref) => {
+const Input = React.forwardRef<HTMLInputElement, InputProps & { icon?: React.ReactNode }>(
```

Comment:
Don't you think it would be better to edit `InputProps` instead since it can be imported from another component?